KOGITO-1229 list redesign

AjayJagan opened the pull request:
- added icons to states
 - added external link icon
 - added link to process details page in process name
 - added popover for error instances

cristianonicolai commented on packages/management-console/server/MockData/types.js:
```diff
@@ -88,12 +88,12 @@ module.exports = typeDefs = gql`
   }
 
   enum ProcessInstanceState {
-    PENDING
```
The actual enum is upper case on Data Index, changing here will probably break the query, see https://github.com/kiegroup/kogito-apps/blob/master/data-index/data-index-service/src/main/resources/schema.graphqls#L73

srambach replied:
I have CSS Magic... we can still display it as "Pending" even if it comes in as "PENDING" by using `text-transform: capitalize;` on that element when we put it into the list. 🧙‍♀️

AjayJagan commented on packages/management-console/src/components/Molecules/DataToolbarComponent/Datatoolbar.css:
```diff
@@ -0,0 +1,14 @@
+
+.pf-c-check__label {
+  text-transform: lowercase;
+}
```
Hey @srambach - I tried adding text-transform:capitalize , but it was still showing every letter in caps. So I did this work around. I am still trying to figure it out, if I am not successful ,please help me out :)

srambach commented:
@AjayJagan it's coming into a better shape :) @cristianonicolai 
- All status icons should be default (black) color except for completed (green) and error (red) as those are.
- Spacing of the Apply Filter button is off (I'll comment on the line to fix)
- The name linked to details beside the external icon linked to the endpoint application needs to be addressed. My comments are in the PR I made to your branch. Hopefully there is a name of the endpoint that we could use as text? If not, another option would be to move that to the kebab. But we need to know a little more about how it's used to say for sure - but the details link next to just the external link icon going to the endpoint cannot stay. :-)  https://github.com/AjayJagan/kogito-apps/pull/2

srambach commented:
OK, I guess it won't let me comment on the line - in DataToolbarComponent.tsx, you can remove the `variant="filter-group"` on line 134:
```
 <DataToolbarGroup variant="filter-group">
```
That's intended to remove the space between multiple filter dropdowns, but the button should be spaced normally.

AjayJagan commented:
> @AjayJagan it's coming into a better shape :) @cristianonicolai
> 
> * All status icons should be default (black) color except for completed (green) and error (red) as those are.
> * Spacing of the Apply Filter button is off (I'll comment on the line to fix)
> * The name linked to details beside the external icon linked to the endpoint application needs to be addressed. My comments are in the PR I made to your branch. Hopefully there is a name of the endpoint that we could use as text? If not, another option would be to move that to the kebab. But we need to know a little more about how it's used to say for sure - but the details link next to just the external link icon going to the endpoint cannot stay. :-)  [AjayJagan#2](https://github.com/AjayJagan/kogito-apps/pull/2)
I have removed the variant = filter groups and changed the color's of the status.
The only thing that's pending is the endpoint. @cristianonicolai  can you please tell us what should be done regarding that
![image](https://user-images.githubusercontent.com/36824134/75742769-a2d87a00-5d34-11ea-8a02-a914b8da03a4.png)

srambach commented on packages/management-console/src/components/Molecules/DataListItemComponent/DataListItemComponent.tsx:
```diff
@@ -389,136 +390,204 @@ const DataListItemComponent: React.FC<IOwnProps> = ({
+                    <ExternalLinkAltIcon />
+                  </Button>
+                </a>
+              </DataListCell>,
```
How about using this to solve the problem of the external icon by itself? And then we also talked about only putting it on the parent, can you do that?
```
              <DataListCell key={1}>
                <Link to={'/ProcessInstances/' + processInstanceData.id} className="pf-u-mr-sm">
                    <strong>{processInstanceData.processName} </strong>
                </Link>
                <Link to={`${processInstanceData.endpoint}/management/processes/${processInstanceData.processId}/instances/${processInstanceData.id}`} target="_blank">
                  Endpoint <ExternalLinkAltIcon />
                </Link>
              </DataListCell>,
```
@AjayJagan @cristianonicolai

AjayJagan replied:
yep, displaying it only on the parent is possible. awaiting @cristianonicolai response.

AjayJagan commented:
> I like the endpoint link. Can you make it go to a new tab? @AjayJagan @cristianonicolai
> Otherwise, approved! Thanks.

Done :+1:

cristianonicolai commented:
@AjayJagan noticed a few things:

- Can you add a mock process with status complete so we can check the UI?
- @srambach  Should the popover close when we click elsewhere? Otherwise we endup with an overlay like in the screenshot below.
- Should buttons in the popover be aligned to the right side? Do we need a close button there?
![Selection_032](https://user-images.githubusercontent.com/570894/76386899-54208500-63b1-11ea-8867-2e86b05530c1.png)
- In the process details page, error state is not showing up
![Selection_033](https://user-images.githubusercontent.com/570894/76387040-afeb0e00-63b1-11ea-8701-f72a2ece87a8.png)

cristianonicolai commented on packages/management-console/server/server.js:
```diff
@@ -50,7 +50,7 @@ app.post(
   controller.callRetrigger
 );
 app.post(
-  '/management/processes/:processId/instances/:processInstanceId',
```
the endpoint was correct, method should be DELETE.

AjayJagan replied:
This change is made in KOGITO 789

cristianonicolai replied:
perhaps just revert the change here then.

AjayJagan replied:
done :)

cristianonicolai replied:
sorry but still showing _/abort_, I guess the idea was to just revert the change on this line.

cristianonicolai commented on packages/management-console/src/components/Organisms/ProcessDetails/ProcessDetails.tsx:
```diff
@@ -12,14 +12,68 @@ import {
 const ProcessDetails: React.FC<IOwnProps> = ({ data, loading }) => {
+  const stateIconCreator = state => {
+    switch (state) {
+      case 'ACTIVE':
```
would it help to use the TypseScript enum here?

AjayJagan commented:
> @AjayJagan we also need to show the error popup on the process details page, when status is in error.

@cristianonicolai  we were adding a view error along with the process management in process details page, So I thought we had to add the pop up instead of the view error. (because the pop contains Skip and Retry which are related to process management)

cristianonicolai commented:
@AjayJagan we can review the details page in another commit, just revert the change in the abort endpoint and we should be good ti go with this PR.
